File: backend/app/schemas/option_config.py
"""
选项配置 Schema
"""
from typing import Optional, List
from pydantic import BaseModel, Field
from datetime import datetime
from app.models.option_config import OptionType


class OptionConfigBase(BaseModel):
    """选项配置基础 Schema"""
    option_type: OptionType = Field(..., description="选项类型")
    value: str = Field(..., max_length=200, description="选项值")
    label: str = Field(..., max_length=200, description="选项标签")
    parent_id: Optional[int] = Field(None, description="父选项ID")
    sort_order: int = Field(0, description="排序顺序")
    is_active: bool = Field(True, description="是否启用")
    description: Optional[str] = Field(None, description="选项描述")


class OptionConfigCreate(OptionConfigBase):
    """创建选项配置 Schema"""
    pass


class OptionConfigUpdate(BaseModel):
    """更新选项配置 Schema"""
    value: Optional[str] = Field(None, max_length=200, description="选项值")
    label: Optional[str] = Field(None, max_length=200, description="选项标签")
    parent_id: Optional[int] = Field(None, description="父选项ID")
    sort_order: Optional[int] = Field(None, description="排序顺序")
    is_active: Optional[bool] = Field(None, description="是否启用")
    description: Optional[str] = Field(None, description="选项描述")


class OptionConfigResponse(OptionConfigBase):
    """选项配置响应 Schema"""
    id: int
    level: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class OptionConfigTree(OptionConfigResponse):
    """选项配置树形结构 Schema"""
    children: Optional[List['OptionConfigTree']] = None


# 允许递归引用
OptionConfigTree.model_rebuild()


class OptionConfigListResponse(BaseModel):
    """选项配置列表响应（树形结构）"""
    items: List[OptionConfigTree]


class MemberMenuVisibilityItem(BaseModel):
    """成员菜单可见性项"""
    menu_key: str = Field(..., description="菜单路径key")
    label: str = Field(..., description="菜单名称")
    is_visible: bool = Field(..., description="成员是否可见")


class MemberMenuVisibilityListResponse(BaseModel):
    """成员菜单可见性列表响应"""
    items: List[MemberMenuVisibilityItem]


class MemberMenuVisibilityUpdate(BaseModel):
    """更新成员菜单可见性请求"""
    menu_key: str = Field(..., description="菜单路径key")
    is_visible: bool = Field(..., description="是否显示给成员")

File: frontend/src/services/optionConfig.ts
/**
 * 选项配置服务
 */
import apiClient from './api'

export type OptionType = 'requirement_direction' | 'product' | 'member_menu_visibility'

export interface OptionConfig {
  id: number
  option_type: OptionType
  value: string
  label: string
  parent_id: number | null
  level: number
  sort_order: number
  is_active: boolean
  description: string | null
  created_at: string
  updated_at: string
  children?: OptionConfig[]
}

export interface OptionConfigCreate {
  option_type: OptionType
  value: string
  label: string
  parent_id?: number | null
  sort_order?: number
  is_active?: boolean
  description?: string | null
}

export interface OptionConfigUpdate {
  value?: string
  label?: string
  parent_id?: number | null
  sort_order?: number
  is_active?: boolean
  description?: string | null
}

export interface OptionConfigListResponse {
  items: OptionConfig[]
}

export interface MemberMenuVisibilityItem {
  menu_key: string
  label: string
  is_visible: boolean
}

export interface MemberMenuVisibilityListResponse {
  items: MemberMenuVisibilityItem[]
}

export const optionConfigService = {
  /**
   * 获取选项配置列表（公开接口，所有用户可访问）
   */
  getOptionConfigs: async (optionType: OptionType): Promise<OptionConfigListResponse> => {
    // apiClient 的响应拦截器已经返回了 response.data，所以这里直接返回 response
    const response = await apiClient.get<OptionConfigListResponse>(
      `/option-configs/${optionType}`
    )
    return response as unknown as OptionConfigListResponse
  },

  /**
   * 获取选项配置列表（管理接口，仅总管可访问，包含禁用项）
   */
  getOptionConfigsAdmin: async (optionType: OptionType): Promise<OptionConfigListResponse> => {
    // apiClient 的响应拦截器已经返回了 response.data，所以这里直接返回 response
    const response = await apiClient.get<OptionConfigListResponse>(
      `/option-configs/admin/${optionType}`
    )
    return response as unknown as OptionConfigListResponse
  },

  /**
   * 创建选项配置
   */
  createOptionConfig: async (data: OptionConfigCreate): Promise<OptionConfig> => {
    // apiClient 的响应拦截器已经返回了 response.data，所以这里直接返回 response
    const response = await apiClient.post<OptionConfig>(
      '/option-configs',
      data
    )
    return response as unknown as OptionConfig
  },

  /**
   * 更新选项配置
   */
  updateOptionConfig: async (id: number, data: OptionConfigUpdate): Promise<OptionConfig> => {
    // apiClient 的响应拦截器已经返回了 response.data，所以这里直接返回 response
    const response = await apiClient.put<OptionConfig>(
      `/option-configs/${id}`,
      data
    )
    return response as unknown as OptionConfig
  },

  /**
   * 删除选项配置（软删除：设置为禁用状态）
   */
  deleteOptionConfig: async (id: number): Promise<void> => {
    await apiClient.delete(`/option-configs/${id}`)
  },

  /**
   * 获取成员菜单可见性配置
   */
  getMemberMenuVisibility: async (): Promise<MemberMenuVisibilityListResponse> => {
    const response = await apiClient.get<MemberMenuVisibilityListResponse>(
      '/option-configs/member-menu-visibility'
    )
    return response as unknown as MemberMenuVisibilityListResponse
  },

  /**
   * 更新成员菜单可见性配置（仅总管）
   */
  updateMemberMenuVisibility: async (menuKey: string, isVisible: boolean): Promise<MemberMenuVisibilityItem> => {
    const response = await apiClient.put<MemberMenuVisibilityItem>(
      '/option-configs/member-menu-visibility',
      { menu_key: menuKey, is_visible: isVisible }
    )
    return response as unknown as MemberMenuVisibilityItem
  },
}

